idea-plugin/src/main/java/com/palantir/gradle/jdks/GradleJdksExternalSystemTaskNotificationListener.java

```diff
@@ -33,6 +33,7 @@ public void onStart(@NotNull ExternalSystemTaskId id, String _workingDir) {
                         || id.getType() == ExternalSystemTaskType.EXECUTE_TASK)) {
             Project project = id.findProject();
             if (project != null) {
+                project.getService(PluginUpdateCheckerService.class).checkPluginVersion();
```

Comment:
I guess when projects are being refreshed _is_ the place to do this as that's when the `.idea/external-dependencies.xml` is being changed.